spec/features/publish/courses/publishing_a_course_spec.rb

```diff
@@ -0,0 +1,117 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+feature "Publishing courses" do
+  before do
+    given_i_am_authenticated_as_a_provider_user
+  end
+
+  scenario "i can publish a course" do
+    and_there_is_a_course_i_want_to_publish
+    when_i_visit_the_course_page
+    and_i_click_the_publish_link
+    then_i_should_see_a_success_message
+    and_the_course_is_published
+  end
+
+  scenario "i can re-publish a course" do
+    given_ive_published_a_course
+    and_i_make_some_new_changes
+    then_i_should_see_the_unpublished_changes_message
+    and_i_should_see_the_publish_button
+  end
+
+  scenario "attempting to publish with errors" do
+    and_there_is_a_course_i_just_created
```

Comment:
can we call out the status of course

```suggestion
    and_there_is_a_draft_course
```

Reply:
done